### PR TITLE
[KEYCLOAK-10694] [KEYCLOAK-10103] RHEL-8 UBI minimal / EAP CD 17 modules / CEKit 3.x support [KEYCLOAK-10124] Define RHEL-8 content set for the Red Hat Single Sign-On 7 OpenShift container image [KEYCLOAK-10556] Drop the MongoDB database driver. Replace MySQL database driver with the MariaDB [KEYCLOAK-10858] Address RH-SSO pod launch failures with MySQL & PostgreSQL DBs, leading to WFLYJCA0069 error

### PR DESCRIPTION
**Pull request by iankko**
[KEYCLOAK-10694] [KEYCLOAK-10103] RHEL-8 UBI minimal / EAP CD 17 modules / CEKit 3.x support
    [KEYCLOAK-10124] Define RHEL-8 content set for the Red Hat Single Sign-On 7 OpenShift container image
    
    * Start using EAP modules for EAP CD 17 / JDK 11 / RHEL-8 UBI support:
      https://github.com/jboss-container-images/jboss-eap-7-openshift-image/commit/cde917a051b65dd748f72ec2b9ab2d130158132f#diff-b56ad46b43eb108be045b7ad559c32b6
      and drop the use of deprecated EAP modules,
    
    * Update CCT and EAP modules to 'master' to pick most fresh
      RHEL-8 UBI / CEKit 3.x changes. Also default to 'microdnf'
      package manager and define RHEL-8 content set,
    
    * Introduce 'overrides' subdirectory as a common location to
      keep examples of various CEKit Overrides files,
    
    * Introduce 'backward-compatibility' subdirectory as a common
      location to keep various CEKit image descriptor files, that
      need to be archived each time a major, backward-incompatible
      change is introduced to CEKit. Archive image file descriptor,
      used to build the image:
      * Using CEKit 2.x,
      * Use yum as the package manager, when the parent image was
        based on RHEL-7 image,
      * Use JBoss EAP modules of version 'CD16' (or earlier) in
        the image
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

--

    [KEYCLOAK-10556] Inherit the original 'os-eap-db-drivers' module
    implementation from EAP_722 release of jboss-eap-modules repository:
    
      https://github.com/jboss-container-images/jboss-eap-modules/tree/c910472e03d6ce281654f171d56c227d493cb900/os-eap-db-drivers
    
    into new 'sso.db.drivers' RH-SSO module.
    
    Also:
    * Drop the MongoDB database driver, and
    * Replace MySQL database driver with the MariaDB one
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

--

    [KEYCLOAK-10858] Address RH-SSO pod launch failures, leading
    to WFLYJCA0069 error with certain combination of DB_SERVICE_PREFIX_MAPPING /
    ${prefix}_DRIVER variables provided to the pod
    
    * Derive XA connection properties & DB driver from the value of
      DB_SERVICE_PREFIX_MAPPING environment variable if provided, to
      avoid this error
    
    [KEYCLOAK-10694] Use MariaDB client instead of the MySQL one in
    RH-SSO OpenShift image based on RHEL-8 UBI minimal image:
    
    * Modify standalone-openshift.xml so MariaDB JDBC driver can be used
      as a functional replacement of the MySQL one,
    
    * Override definitions of certain TX & XA datasource creation /
      processing routines from common JBoss EAP modules
      ('os-eap-datasource' and 'os-eap7-launch' one) to:
    
      -- Replace any request for MySQL DB driver with a request for the MariaDB
         one,
    
      -- Forbid JDBC requests (via DB_SERVICE_PREFIX_MAPPING or ${prefix}_DRIVER
         environment variables) to use of any other DB drivers, than just MariaDB
         and PostgreSQL ones. Instruct users to install other JDBC drivers via
         the usual JBoss EAP Runtime Artifacts Datasources mechanism
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>
--

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by iankko** [comment]
@hmlnarik @slaskawi Refactored the original PR to preserve backward-compatibility (reproducible image (re)-builds for RH-SSO CD image versions of 6.0 and earlier).

Confirmed, the image builds correctly on top of RHEL-8 UBI minimal image. Also the MongoDB driver gets dropped correctly, and the MySQL DB driver is replaced with the MariaDB one.

There's only one issue left (probably related with the definition of the respective MySQL / MariaDB datasource) when deploying the image from templates. The RH-SSO pod fails to start due to some MySQL / MariaDB warning.

This will be investigated further within work at [KEYCLOAK-10556](https://issues.jboss.org/browse/KEYCLOAK-10556) -- assuming the fix will involve the changes to [OS EAP datasource module](https://github.com/jboss-container-images/jboss-eap-modules/tree/master/os-eap-datasource/1.0). Once I have such a fix / change ready, we would use version `2.0` of that module (IOW one-liner fix for the OpenShift image repository).

**Comment by hmlnarik** [comment]
@iankko What exactly is the warning? Furthermore, could this PR be split into two? I'm a bit worried about KEYCLOAK-10556 and if possible, rather have the UBI8 update separately so that it is not blocked by mongodb extraction.

**Comment by iankko** [comment]
> @iankko What exactly is the warning?

Reported now under separate new [KEYCLOAK-10858](https://issues.jboss.org/browse/KEYCLOAK-10858) JIRA (including logs for both cases, starting the pod using MySQL and the PostgreSQL templates -- the issue manifestates in both cases, just the symptoms differ slightly).

> Furthermore, could this PR be split into two? I'm a bit worried about KEYCLOAK-10556 and if possible, rather have the UBI8 update separately so that it is not blocked by mongodb extraction.

Sure, it can be split. But what should be provided for the database module? Just the PostgreSQL driver for now? Or no DB module at all? (the old RHEL-7 one can't be used due to rebase to RHEL-8 image)

File: modules/sso/config/launch/setup/73/added/standalone-openshift.xml
Comment (hmlnarik) [inline modules/sso/config/launch/setup/73/added/standalone-openshift.xml]: Needs to be `org.mariadb.jdbc.MariaDbDataSource` per https://mariadb.com/kb/en/library/about-mariadb-connector-j/#optional-jdbc-classes

Comment (iankko) [reply modules/sso/config/launch/setup/73/added/standalone-openshift.xml]: @hmlnarik Great catch, thanks! Updated / fixed & squashed into previous change.

**Comment by iankko** [comment]
JFTR, please DNM yet (the MySQL / MariaDB KEYCLOAK-108585 WIP issue needs to be addressed first). Will drop the WIP status, once it's clear for review / merge.

**Comment by iankko** [comment]
OSBS information (branch, build target & component name) updated now too (see 4-th commit). This is ready for review.

**Review by hmlnarik — APPROVED** [review]
Thanks @iankko !

**Pull request merged**
